### PR TITLE
Update buildRegex function

### PR DESCRIPTION
**Pull request by raych1**
Support wildcard match in case of having parameter in the beginning of the path;
Revert the change in validateOption check.

### CODE REVIEW

File: lib/types/path.js [hunk 1]
```diff
@@ -140,12 +140,12 @@ function Path (api, path, definition, definitionFullyResolved, pathToDefinition,
  * @param {bool} isCaseSensitive - specifies if to use case sensitive comparison or not
  * @returns {object} The pathToRegexp object
  */
-function buildRegex (hostTemplate, basePathPrefix, path, isCaseSensitive) {
+ function buildRegex (hostTemplate, basePathPrefix, path, isCaseSensitive) {
```
Comment (PhoenixHe-NV) [inline lib/types/path.js]: format

File: lib/validation/custom-zschema-validators.js [hunk 2]
```diff
@@ -205,7 +205,7 @@ function shouldSkipValidate (options, errors) {
     Array.isArray(options.includeErrors) &&
     options.includeErrors.length > 0 &&
     !errors.some(function (err) {
-      return !options.includeErrors.includes(err);
```
Comment (PhoenixHe-NV) [inline lib/validation/custom-zschema-validators.js:208]: Why this change?

**Review by PhoenixHe-msft — APPROVED** [review]

File: lib/validation/custom-zschema-validators.js (hunk 2 above)
Comment (raych1) [reply lib/validation/custom-zschema-validators.js:208]: This is to revert the previous change to flip the logic to suppress the error in validation because it depends on the z-schema change which isn't done yet.
